main_test.go

```diff
@@ -5,77 +5,140 @@ import (
 	"net/http"
 	"net/http/httptest"
 	"os"
+	"strings"
 	"sync/atomic"
 	"testing"
 	"time"
 )
 
-func testServer(t *testing.T) *httptest.Server {
-	var attempt int32
-	paths := make(map[string]func(http.ResponseWriter))
-	paths["/ok"] = func(w http.ResponseWriter) {
-		fmt.Fprintf(w, "42")
+func TestDownload_Error(t *testing.T) {
+	timeout := 250 * time.Millisecond
+	for _, tc := range []struct {
+		desc string
+		proc func(w http.ResponseWriter)
+	}{
+		{"failure", func(w http.ResponseWriter) { w.WriteHeader(http.StatusBadRequest) }},
+		{"timeout", func(w http.ResponseWriter) { time.Sleep(10 * timeout) }},
```

Comment:
Does it worth it to think about a test scenario that timeouts due to `d.client.Timeout` and another one that fails because of the context created with timeout?

(this can be in another PR btw)

Reply:
* The HTTP client is private, thus we are already testing its timeout. 
* Added the test that fails if a user-specified timeout is shorter than the timeout.